### PR TITLE
Cards can be played by clicking, solving #35

### PR DESCRIPTION
**Pull request by Masclins**
This lets playing by clicking card and row afterwards.

It also highlights the selected card (that's going to be played).

Since this is really tied to #16, I'd like a review before moving into it, so I go there with the code as clear as possible.

### CODE REVIEW

File: frontend/game/board/board.js [hunk 1]
```diff
@@ -32,5 +32,22 @@ define(function (require) {
         },
         allowDrop: drop.allow,
         dropOnRow: drop.onRow,
+        clickRow(event) {
+            if (helper.clickedCard != -1) {
```
Comment (kenan-rhoton) [inline frontend/game/board/board.js]: Please extract the playCard logic so this and drop.onRow can share a function

File: frontend/game/hand.js [hunk 2]
```diff
@@ -2,6 +2,23 @@
...
+    function clearBackground() {
+        document.querySelectorAll(".card").forEach(function(element) {
+            element.style.background = "white"
+        });
+    }
+
+    function clickCard(card) {
+        clearBackground();
+        if (helper.clickedCard == card) {
+            helper.clickedCard = -1;
```
Comment (kenan-rhoton) [inline frontend/game/hand.js]: please use `undefined` instead of `-1`

File: frontend/game/helper.js [hunk 3]
```diff
@@ -2,17 +2,33 @@
...
         },
-        clickedCard: -1,
-        gameID: params.get("gameID"),
-        playerID: params.get("playerID"),
+        clickedCard: undefined,
+        gameID: gameID,
+        playerID: playerID,
         baseCard: document.getElementById("card-template")
                 .content.querySelector(".card"),
+        playCard(rownum, cardindex) {
```
Comment (kenan-rhoton) [inline frontend/game/helper.js]: I think this should be in board/play.js rather than helper 🤔

**Comment by kenan-rhoton** [comment]
Ok, I see 2 options now: either we move `play.js` to be inside `board` or we include the `hand` functions that have to do with playing a card in it.

There's also a third option of having a new `game/play` directory with a file for the `board` functions and another for the `hand` functions 🙃

**Comment by Masclins** [comment]
Tests pass in my computer but seems not to pass for Travis...

**Comment by Masclins** [comment]
About what to do with `play.js`:

1. *`play.js` inside `board/`*: This feels weird to me. Playing a card is not something specific to the board.
2. *Move some functions in `hand.js` to `play.js`*: I assume you talk about clicking and dragging a card. This seems somewhat ok, though I'm not sure if it really makes anything clearer. And then probably should move the equivalents from `board.js`. And by doing so `play.js` would end up being a mess.
3. *Having `game/play/` with `play.js`, `hand.js` and `board.js`*: If we have to choose one, this has the most sense to me. My only question is if we really want to divide files that much.

So, in my oppinion, we either leave it as it is or do the third option.

**Comment by kenan-rhoton** [comment]
Go with option three then, I will fix travis in parallel

File: frontend/game/index.html [hunk 4]
```diff
@@ -11,7 +11,7 @@
         <div id="app" class="text-center container">
             <h1 class="display-1">
                 <a href="/">
-                    Troll Card Game
+                    TCG
```
Comment (Masclins) [inline frontend/game/index.html]: I really hope you are ok with this.
It's too NSFW to have **Troll Card Game** on screen when testing something.

File: frontend/game/hand.js [hunk 5]
```diff
@@ -28,7 +13,7 @@ define(function(require) {
                 var newCard = helper.baseCard.cloneNode(true);
                 newCard.innerHTML = cardInHand["power"];
                 newCard.setAttribute("index", index);
-                newCard.addEventListener('click', function(){clickCard(newCard)});
+                newCard.addEventListener('click', function(){play.clickCard(newCard)});
```
Comment (Masclins) [inline frontend/game/hand.js]: I have the feeling this could be prettier, but don't know how.

File: frontend/game/play/play.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 "use strict";
 
-define(function() {
-    var helper = require('./helper.js');
+define(function(require) {
```
Comment (Masclins) [inline frontend/game/play/play.js]: Maybe this is why tests failed.

File: frontend/game/hand.js (hunk 5 above)
Comment (kenan-rhoton) [reply frontend/game/hand.js]:
```javascript
newCard.addEventListener('click', () => play.clickCard(newCard) );
```
maybe?

File: frontend/game/play/play.js (hunk 6 above)
Comment (kenan-rhoton) [reply frontend/game/play/play.js]: nah, that's a separate issue I have to fix

File: frontend/game/index.html (hunk 4 above)
Comment (kenan-rhoton) [reply frontend/game/index.html]: good idea

File: frontend/game/hand.js (hunk 5 above)
Comment (Masclins) [reply frontend/game/hand.js]: I saw this alternative on SE.
But it seems equally ugly + harder to understand.

**Pull request merged**